crawl4ai/processors/pdf/__init__.py

```
@@ -124,7 +125,13 @@ def scrap(self, url: str, html: str, **params) -> ScrapingResult:
         finally:
             # Cleanup temp file if downloaded
             if url.startswith(("http://", "https://")):
-                Path(pdf_path).unlink(missing_ok=True)
+                try:
+                    Path(pdf_path).unlink(missing_ok=True)
+                    if pdf_path in self._temp_files:
+                        self._temp_files.remove(pdf_path)
+                except Exception as e:
```

Ruff (0.14.0)

Do not catch blind exception: `Exception`

(BLE001)